### PR TITLE
Heat unfreezes things

### LINKED ISSUE #21330: Ice never fucking melts
I've seen fires burning over ice, but nothing ever seems to get it to melt.

### PR DESCRIPTION
**Pull request by Cyberboss**
My goal as a spessman is to make freon great ~~again~~

Fixes #21330

:cl: Cyberboss
fix: Frozen things will now unfreeze above 0C
/:cl:

A few objs may randomly stay frozen for no reason apparently, can be fixed by moving them (Open to speculation as to why)
Frozen things not on open turfs (air alarms, posters, etc...) are SoL

### CODE REVIEW

**Comment by MrAlphonzo** [comment]
Goof couldn't even bother to clean up his own mess.

**Comment by Iamgoofball** [comment]
:+1:

@MrAlphonzo dicks

File: code/__HELPERS/icons.dm [hunk 1]
```diff
@@ -1013,27 +1013,27 @@ var/list/freeze_item_icons = list()
 /atom/proc/freeze_icon_index()
 	return "\ref[initial(icon)]-[initial(icon_state)]"
 
-/obj/proc/make_frozen_visual(var/obj/F)
-	if(!F.is_frozen && (initial(icon) && initial(icon_state)))
+/obj/proc/make_frozen_visual
```
Comment (Iamgoofball) [inline code/__HELPERS/icons.dm]: forgot the ()

File: code/game/turfs/open.dm [hunk 2]
```diff
@@ -228,6 +231,8 @@
 				wet_time = max(0, wet_time-10)
 			if(T0C + 100 to INFINITY)
 				wet_time = 0
+	else if (GetTemperature() > BODYTEMP_COLD_DAMAGE_LIMIT)	//seems like a good place
+		MakeDry(TURF_WET_PERMAFROST)
```
Comment (Cyberboss) [inline code/game/turfs/open.dm]: There's literally no way currently right now, so this stays until there is another.

Also it only degrades to ice

Comment (Iamgoofball) [inline code/game/turfs/open.dm]: uh, there is totally a way

use a crowbar

Comment (lzimann) [inline code/game/turfs/open.dm]: yes because using a crowbar to make ice disappear makes way more sense than heating up the place

**Comment by MrAlphonzo** [comment]
Its true tho

File: code/game/turfs/open.dm [hunk 3]
```diff
@@ -208,6 +208,9 @@
 	if(wet_time > MAXIMUM_WET_TIME)
 		wet_time = MAXIMUM_WET_TIME
 	if(wet == TURF_WET_ICE && air.temperature > T0C)
+		for(var/obj/O in contents)
+			if(O.is_frozen)
+				O.make_unfrozen()
```
Comment (XDTM) [inline code/game/turfs/open.dm]: what about frozen items that are not on frozen tiles anymore

File: code/game/turfs/open.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/game/turfs/open.dm]: it's not about logic you backwards fuckin redneck it's about gameplay

File: code/game/turfs/open.dm (hunk 3 above)
Comment (Cyberboss) [inline code/game/turfs/open.dm]: https://github.com/tgstation/tgstation/pull/22054/files#diff-2939b84f8f6de56f3b4a70ef33122366R138See the

File: code/game/turfs/open.dm (hunk 2 above)
Comment (Cyberboss) [inline code/game/turfs/open.dm]: It should frankly look cooler when atmos brings in a heater and the entire place starts melting

File: code/game/turfs/open.dm (hunk 3 above)
Comment (XDTM) [inline code/game/turfs/open.dm]: will this fire if you carry them in hand?

File: code/game/turfs/open.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/game/turfs/open.dm]: fair point

Comment (Iamgoofball) [inline code/game/turfs/open.dm]: @Mindustry re-apply tile

**Comment by Armhulen** [comment]
Does this fix the very easy crowbar fixing?

**Comment by Cyberboss** [comment]
Crowbaring a tile hasan't changed

File: code/game/turfs/open.dm (hunk 3 above)
Comment (Cyberboss) [inline code/game/turfs/open.dm]: no but it will when you drop them i think

**Comment by Cyberboss** [comment]
Just tested btw. It takes 2 heaters around 5 minutes to unfreeze the bar after breaking open a canister. So it's not really powergamey

File: code/game/turfs/open.dm (hunk 2 above)
Comment (ShadowDeath6) [inline code/game/turfs/open.dm]: Could have a layer of water be on the tile after melting, the only way for ice to instantly evaporate is if you used plasma fire or something

**Comment by Cyberboss** [comment]
It currently degrades from permafrost(nonstun slide) to ice(regular slip)
to nothing. Yeah that's a good idea

File: code/__HELPERS/icons.dm [hunk 4]
```diff
@@ -1013,21 +1013,27 @@ var/list/freeze_item_icons = list()
...
-		F.icon = P
-		F.name = "frozen [F.name]"
-		F.is_frozen = TRUE
+		icon = P
+		name = "frozen [name]"
+		is_frozen = TRUE
 		return
-	return
\ No newline at end of file
+	return
```
Comment (ChangelingRain) [inline code/__HELPERS/icons.dm]: trailing returns, brrr

**Pull request merged**